cmd/lnd/main.go

```diff
@@ -30,11 +30,12 @@ func main() {
 		// Help was requested, exit normally.
 		os.Exit(0)
 	}
+	implCfg := loadedConfig.ImplementationConfig(shutdownInterceptor)
 
 	// Call the "real" main in a nested manner so the defers will properly
 	// be executed in the case of a graceful shutdown.
 	if err = lnd.Main(
-		loadedConfig, lnd.ListenerCfg{}, shutdownInterceptor,
+		loadedConfig, lnd.ListenerCfg{}, implCfg, shutdownInterceptor,
```

Comment:
How is it envisioned that a new caller swaps out some of these interfaces? By create a new wrapper struct that implements some of the newly defined builder methods, which then intercepts them (to take precedence) from the default implementations?

Reply:
Or perhaps via the functional arg additions that would be passed in as additional arguments into `ImplementationConfig`? (similar to how things are outlined in that original brain storm issue)

Reply:
Yes, those are both good examples. Or just implement the interfaces fully. For now we use the "embed and intercept" way in https://github.com/lightningnetwork/lnd/pull/5689. But can rework this again later if needed. Just having everything behind interfaces should make this flexible enough for now I think.